src/PFire.Core/Protocol/MessageTypeFactory.cs

```diff
@@ -60,7 +74,7 @@ public Type GetMessageType(XFireMessageType messageType)
 
             if (!_messages.ContainsKey(sType))
             {
-                throw new UnknownMessageTypeException(messageType);
+                return _messages[999].GetType();
```

Comment:
What's 999? I prefer we continue to throw an exception here not return an unknown message